crates/oxc_parser/src/js/expression.rs

```diff
@@ -147,6 +147,10 @@ impl<'a> Parser<'a> {
     fn parse_primary_expression(&mut self) -> Result<Expression<'a>> {
         let span = self.start_span();
 
+        if self.at(Kind::At) {
+            self.eat_decorators()?;
+        }
+
         // AsyncFunctionExpression
         // AsyncGeneratorExpression
         if self.at_function_with_async() {
```

Comment:
For consistency with declarations. Should probably add an early-error pass to check decorators in wrong kinds of expressions/declarations in the future.